addons/hr_expense/models/hr_expense.py

```diff
@@ -1289,13 +1289,17 @@ def _prepare_payment_vals(self):
 
     def _prepare_bill_vals(self):
         self.ensure_one()
+        if self.payment_mode == 'company_account':
+            partner_id = self.employee_id.sudo().address_home_id.commercial_partner_id.id
+        else:
+            partner_id = self.employee_id.id
```

Comment:
Did you test the fix because I'm surprised it works.
We discussed it a bit with JUAL and unless I'm mistaken, `self.employee_id` isn't a `res.partner` and thus it is probably working by chance. :/